docs/writing-tests/testharness-tutorial.md

```diff
@@ -0,0 +1,390 @@
+  this, we've used `assert_unreached` *in the fulfillment handler*.
+  `assert_unreached` is a testharness.js utility function which always throws
+  an error. With this in place, if fetch does *not* produce an error, then this
+  subtest will fail.
+
+  We've moved the assertion about the cookie to the rejection handler. We also
+  switched from `assert_true` to `assert_false` because the test should only
+  pass if the cookie is *not* set. It's a good thing we have the cleanup logic
+  in the previous subtest, right?
+
+If you run the test in your browser now, you can expect to see both tests
+reported as passing with their distinct names.
+
+![](../assets/testharness-tutorial-test-screenshot-2.png "screen shot of testharness.js reporting the test results")
+
+## Verifying our work
+
+We're done writing the test, but we should make sure it fits in with the rest
+of WPT before we submit it.
+
+[The lint tool](lint-tool) can detect some of the common mistakes people make
+when contributing to WPT. You enabled it when you [configured your system to
+work with WPT](../running-tests/from-local-system). To run it, open a
+command-line terminal, navigate to the root of the WPT repository, and enter
+the following command:
+
+    python ./wpt lint fetch/api/basic
+
+If this recognizes any of those common mistakes in the new files, it will tell
+you where they are and how to fix them. If you do have changes to make, you can
+run the command again to make sure you got them right.
+
+Now, we'll run the test using the automated test runner. This is important for
+testharness.js tests because there are subtleties of the automated test runner
+which can influence how the test behaves. That's not to say your test has to
+pass in all browsers (or even in *any* browser). But if we expect the test to
+pass, then running it this way will help us catch other kinds of mistakes.
+
+The tools support running the tests in many different browsers. We'll use
+Firefox this time:
+
+    python ./wpt run firefox fetch/api/basic/set-cookie.html
+
+We expect this test to pass, so if it does, we're ready to submit it. If we
+were testing a web platform feature that Firefox didn't support, we would
+expect the test to fail instead.
+
+There are a few problems to look out for in addition to passing/failing status.
+The report will describe fewer tests than we expect if the isn't run at all.
+That's usually a sign of a formatting mistake, so you'll want to make sure
+you've used the right file names and metadata. Separately, the web browser
+might crash. That's often a sign of a browser bug, so you should consider
+reporting it to the browser's maintainers!
+
+## Submitting the test
+
+First, let's stage the new files for committing:
+
+    $ git add fetch/api/basic/set-cookie.asis
+    $ git add fetch/api/basic/set-cookie.html
+
+We can make sure the commit has everything we want to submit (and nothing we
+don't) by using `git diff`:
```

Comment:
Personally I still trip over this when reading — the way it’s worded (as I noted in a review comment about the same language in the other test tutorial). But again I don’t feel strongly about it needing to be changed.